lib/bolt/applicator.rb

```diff
@@ -22,14 +22,22 @@ def catalog_apply_task
       Task.new('apply_catalog', [impl], 'stdin')
     end
 
-    def compile(target, ast)
+    def compile(target, ast, plan_vars)
+      trusted = {
+        'certname' => target.host,
+        'hostname' => target.host,
+        'domain' => target.host,
```

Comment:
Using host as domain seems wrong... Should we try to cut it out of the hostname or maybe omit it altogether?

Reply:
I think I'll try using https://github.com/puppetlabs/puppet/blob/master/lib/puppet/context/trusted_information.rb to generate this.